### PR TITLE
Code Refactoring for optimization

### PR DESCRIPTION
**Pull request by deye9**

### CODE REVIEW

**Review by sfagade — CHANGES_REQUESTED** [review]
Can you please make next PR smaller?

File: libraries/src/main/java/com/foodie/kafka/config/KafkaConfig.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package com.foodie.kafka.config;
+
+// import static org.apache.kafka.streams.StreamsConfig.APPLICATION_ID_CONFIG;
```
[the remark this answers is not in the document]
Comment (deye9) [reply libraries/src/main/java/com/foodie/kafka/config/KafkaConfig.java:3]: Still working on this.

File: libraries/src/main/java/com/foodie/kafka/service/KafkaConsumerService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public List<Object> getTopicMessages(String topic) {
+
+        consumer.subscribe(Collections.singletonList(topic));
+        List<Object> messageList = new ArrayList<>();
+
+        do {
+            ConsumerRecords<String, List<Object>> records = consumer.poll(Duration.ofMillis(100));
+
+            for (ConsumerRecord<String, List<Object>> record : records) {
+                System.out.printf("offset = %d, key = %s, value = %s%n", record.offset(),
```
[the remark this answers is not in the document]
Comment (deye9) [reply libraries/src/main/java/com/foodie/kafka/service/KafkaConsumerService.java:29]: Error from my side

File: libraries/src/main/java/com/foodie/kafka/config/KafkaConfig.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    final KafkaProperties kafkaProperties;
+
+    public KafkaConfig(KafkaProperties kafkaProperties) {
+        this.kafkaProperties = kafkaProperties;
+    }
+
+    @Bean
+    KafkaAdmin kafkaAdmin() {
+        KafkaAdmin kafkaAdmin = new KafkaAdmin(kafkaProperties.buildAdminProperties());
```
[the remark this answers is not in the document]
Comment (deye9) [reply libraries/src/main/java/com/foodie/kafka/config/KafkaConfig.java:65]: Working on the fix

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java [hunk 4]
```diff
@@ -26,11 +25,14 @@
 
 @Slf4j
 @RestController
-@RequiredArgsConstructor
 @RequestMapping("/api/v1/auth")
 public class AuthenticationController {
 
-    private final AuthenticationService service;
+    private AuthenticationService service;
```
[the remark this answers is not in the document]
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:31]: It was breaking the code

File: services/user-management/src/main/java/com/foodie/user/service/RoleService.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.springframework.stereotype.Service;
+
+import com.foodie.FoodieBaseRepository;
+import com.foodie.FoodieBaseService;
+import com.foodie.user.model.Role;
+import com.foodie.user.repositories.RoleRepository;
+
+@Service
+public class RoleService implements FoodieBaseService<Role, UUID> {
```
[the remark this answers is not in the document]
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/service/RoleService.java:13]: Magic

**Pull request merged**